framework/test/jest/integration/specs/modules/chain/components/storage/entities/Account/decreaseFieldBy.spec.js

```diff
@@ -0,0 +1,162 @@
+
+		storage.entities.Account.extendDefaultOptions({
+			limit: constants.ACTIVE_DELEGATES,
+		});
+	});
+
+	afterAll(async () => {
+		await pgHelper.cleanup();
+	});
+
+	beforeEach(async () => {
+		await db.query('DELETE FROM mem_accounts');
+	});
+
+	describe('Given arguments ({publicKey}, "producedBlocks", "1", tx)', () => {
+		it('should decrease "producedBlocks" property of each given account by "1"', async () => {
+			// Arrange
+			const account = {
+				address: 'delegateAddress',
+				isDelegate: 0,
+				publicKey:
+					'399a7d14610c4da8800ed929fc6a05133deb8fbac8403dec93226e96fa7590ee',
+				balance: 100,
+				producedBlocks: 5,
+			};
+
+			const expectedValue = 4;
+
+			await pgHelper.createAccount(account);
+
+			// Act
+			await db.tx(async tx => {
+				await storage.entities.Account.decreaseFieldBy(
+					{
+						publicKey: account.publicKey,
+					},
+					'producedBlocks',
+					'1',
+					tx,
+				);
+			});
+
+			const updatedAccount = await pgHelper.getAccountByPublicKey(
+				account.publicKey,
+			);
+
+			// Assert
+			expect(updatedAccount.producedBlocks).toEqual(expectedValue);
+		});
+	});
+
+	describe('Given arguments ({publicKey_in: []}, "missedBlocks", "1", tx)', () => {
+		it('should decrease "missedBlocks" property of each account by 1', async () => {
+			// Arrange
+			const account = {
+				address: 'delegateAddress',
+				isDelegate: 0,
+				publicKey:
+					'399a7d14610c4da8800ed929fc6a05133deb8fbac8403dec93226e96fa7590ee',
+				balance: 100,
+				missedBlocks: 5,
+			};
+
+			const account2 = {
+				address: 'delegateAddress2',
+				isDelegate: 0,
+				publicKey:
+					'abca7d14610c4da8800ed929fc6a05133deb8fbac8403dec93226e96fa7590aa',
+				balance: 100,
+				missedBlocks: 10,
+			};
+
+			const expectedValue1 = 4;
+			const expectedValue2 = 9;
+
+			await pgHelper.createAccount(account);
+			await pgHelper.createAccount(account2);
+
+			// Act
+			await db.tx(async tx => {
+				await storage.entities.Account.decreaseFieldBy(
+					{
+						publicKey_in: [account.publicKey, account2.publicKey],
+					},
+					'missedBlocks',
+					'1',
+					tx,
+				);
+			});
+
+			const updatedAccount1 = await pgHelper.getAccountByPublicKey(
+				account.publicKey,
+			);
+			const updatedAccount2 = await pgHelper.getAccountByPublicKey(
+				account2.publicKey,
+			);
+
+			// Assert
+			expect(updatedAccount1.missedBlocks).toEqual(expectedValue1);
+			expect(updatedAccount2.missedBlocks).toEqual(expectedValue2);
+		});
+	});
+
+	describe('Given arguments ({publicKey}, "voteWeight", "Number", tx)', () => {
+		it('should decrease "voteWeight" property of each given account by given value', async () => {
+			// Arrange
+			const account = {
+				address: 'delegateAddress',
+				isDelegate: 0,
+				publicKey:
+					'399a7d14610c4da8800ed929fc6a05133deb8fbac8403dec93226e96fa7590ee',
+				balance: 100,
+				voteWeight: 200,
+			};
+
+			// @todo check why this is coming string form db.
```

Comment:
Has this been addressed?

Reply:
@michielmulders I'll address that after having the STM for integration test helpers.